src/inatinqperf/utils/profiler.py

```diff
@@ -72,10 +169,95 @@ def __exit__(
             "profiler": "builtin",
         }
 
+        container_metrics = self._summarise_containers()
+        if container_metrics:
+            self.metrics["containers"] = container_metrics
+
         ts = datetime.datetime.now(datetime.UTC).strftime("%Y%m%dT%H%M%SZ")
         path = self.results_dir / f"step-{self.step}-{ts}.json"
 
         with path.open("w", encoding="utf-8") as f:
             json.dump(self.metrics, f, indent=2)
 
         logger.info(f"[PROFILE] {self.metrics}")
+
+    def register_container(self, container: Container | str) -> None:
```

Comment:
Is this a method that can be called by a user of the instance of the class or should it only be called by `__init__`? If only by `__init__`, consider communicating that it is private by prefixing the name with an underscore (`_register_container`).